gtests/net/packetdrill/run_packet.c

```diff
@@ -1248,14 +1248,18 @@ static int verify_outbound_live_tcp_options(
 	/* TCP options are expected to be a deterministic order. */
 	while (a_opt != NULL || s_opt != NULL) {
 		if (a_opt == NULL || s_opt == NULL ||
-		    a_opt->kind != s_opt->kind) {
+		    (a_opt->kind & ~TCPOPT_WILDCARD) !=
```

Comment:
This seems like it would ignore cases where the actual packet has bit 0x80 set for some reason; AFAICT this would indicate a kernel bug, and is not something we should ignore.

There is also the case for an actual TCP option number that sets the 0x80 bit:
  #define TCPOPT_EXP		254	/* Experimental */

If the script says the option type is supposed to be TCPOPT_EXP then we should check that all bits really match TCPOPT_EXP.

Reply:
I would suggest tightening this up with something like:

if (a_opt == NULL || s_opt == NULL ||
    (a_opt->kind != s_opt->kind &&
     !(s_opt->kind == (TCPOPT_WILDCARD | TCPOPT_WINDOW) &&
        a_opt->kind == TCPOPT_WINDOW)) {